linux_os/guide/system/auditing/auditd_configure_rules/audit_login_events/audit_rules_login_events.rule

```diff
@@ -1,6 +1,6 @@
 documentation_complete: true
 
-title: 'Record Attempts to Alter Logon and Logout Events'
+title: 'Record Attempts to Alter Logon and Logout Events via auditd Configuration'
```

Comment:
Let's not add this since no other rule does `via auditd Configuration`

Reply:
The point of this change was to make titles unique. Per other files in this directory, perhaps you'd be more okay with "Record Attempts to Alter Logon and Logout Events - auditd"?

If not this, then we should change https://github.com/cipherboy/scap-security-guide/blob/d155e5c67f15eb005b0443e30b222fa114b78bee/linux_os/guide/system/auditing/auditd_configure_rules/audit_login_events/group.yml#L3 as it is a title conflict.

Reply:
@cipherboy title's do not have to be unique from groups. In this case, the rule and group are the same because login/logout events were not broken out until RHEL7, so keep the original title as is. Also by changing the title you are essentially saying, "Record Attempts to Alter Logon and Logout Events for auditd"; therefore, the title change is incorrect. In addition, adding "via auditd configuration" is essentially alluding to the fact that there are other ways of auditing login/logout events which is also incorrect.

Reply:
@redhatrises I think, within a product, a title should be a unique identifier of an object (even across group, benchmark, variable, or rule) because they are displayed to the end user. If a group and a rule share the same title, it is ambiguous which is being referred to when referred to by title. Since e.g., `scap-workbench` displays this title, the user might expect to disable the rule and have it disable the group and visa versa.


There are two main cases of duplicated titles. What follows is a complete list of all 24 duplicated titles in the entire project grouped by the two categories.

### Copy and Paste Errors (14)

In some cases, these titles are simply wrong due to copy and paste errors:

- REUSED TITLE (Disallow creating symlinks to a file you not own): `debian8/guide/system/fs-part/fs-restrict/sysctl_fs_protected_hardlinks.rule` and `debian8/guide/system/fs-part/fs-restrict/sysctl_fs_protected_symlinks.rule`
- REUSED TITLE (Configure Accepting IPv6 Redirects By Default): `rhel6/guide/system/network/network-ipv6/configuring_ipv6/disabling_ipv6_autoconfig/sysctl_net_ipv6_conf_all_accept_redirects.rule` and `rhel6/guide/system/network/network-ipv6/configuring_ipv6/disabling_ipv6_autoconfig/sysctl_net_ipv6_conf_default_accept_redirects.rule`
- REUSED TITLE (Configure Accepting IPv6 Router Advertisements): `rhel6/guide/system/network/network-ipv6/configuring_ipv6/disabling_ipv6_autoconfig/sysctl_net_ipv6_conf_all_accept_ra.rule` and `rhel6/guide/system/network/network-ipv6/configuring_ipv6/disabling_ipv6_autoconfig/sysctl_net_ipv6_conf_default_accept_ra.rule`
- REUSED TITLE (Configure Kernel Parameter for Accepting Source-Routed Packets for All Interfaces): `rhel6/guide/system/network/network-ipv6/configuring_ipv6/disabling_ipv6_autoconfig/sysctl_net_ipv6_conf_all_accept_source_route.rule` and `rhel6/guide/system/network/network-kernel/network_host_and_router_parameters/sysctl_net_ipv4_conf_all_accept_source_route.rule`
- REUSED TITLE (Use vsftpd to Provide FTP Service if Necessary): `rhel6/guide/services/ftp/ftp_configure_vsftpd/group.yml` and `rhel6/guide/services/ftp/ftp_use_vsftpd/group.yml`
- REUSED TITLE (Disable DHCP Server): `rhel6/guide/services/dhcp/dhcp_server_configuration/group.yml` and `rhel6/guide/services/dhcp/disabling_dhcp_server/group.yml`
- REUSED TITLE (Action for auditd to take when disk space just starts to run low): `linux_os/guide/system/auditing/configure_auditd_data_retention/var_auditd_space_left_action.var` and `linux_os/guide/system/auditing/configure_auditd_data_retention/var_auditd_admin_space_left_action.var`
- REUSED TITLE (Configure Kernel Parameter for Accepting Source-Routed Packets for All Interfaces): `linux_os/guide/system/network/network-ipv6/configuring_ipv6/disabling_ipv6_autoconfig/sysctl_net_ipv6_conf_all_accept_source_route.rule` and `linux_os/guide/system/network/network-kernel/network_host_and_router_parameters/sysctl_net_ipv4_conf_all_accept_source_route.rule`
- REUSED TITLE (Configure Accepting IPv6 Redirects By Default): `linux_os/guide/system/network/network-ipv6/configuring_ipv6/disabling_ipv6_autoconfig/sysctl_net_ipv6_conf_all_accept_redirects.rule` and `linux_os/guide/system/network/network-ipv6/configuring_ipv6/disabling_ipv6_autoconfig/sysctl_net_ipv6_conf_default_accept_redirects.rule`
- REUSED TITLE (Configure Accepting IPv6 Router Advertisements): `linux_os/guide/system/network/network-ipv6/configuring_ipv6/disabling_ipv6_autoconfig/sysctl_net_ipv6_conf_all_accept_ra.rule` and `linux_os/guide/system/network/network-ipv6/configuring_ipv6/disabling_ipv6_autoconfig/sysctl_net_ipv6_conf_default_accept_ra.rule`
- REUSED TITLE (Use vsftpd to Provide FTP Service if Necessary): `linux_os/guide/services/ftp/ftp_configure_vsftpd/group.yml` and `linux_os/guide/services/ftp/ftp_use_vsftpd/group.yml`
- REUSED TITLE (Login Banner Verbiage): `linux_os/guide/services/http/securing_httpd/httpd_secure_content/var_web_login_banner_text.var` and `linux_os/guide/system/accounts/accounts-banners/login_banner_text.var`
- REUSED TITLE (Configure PHP Securely): `linux_os/guide/services/http/securing_httpd/httpd_configure_php_securely/group.yml` and `linux_os/guide/services/http/securing_httpd/httpd_configure_perl_securely/group.yml`
- REUSED TITLE (Disable DHCP Server): `linux_os/guide/services/dhcp/dhcp_server_configuration/group.yml` and `linux_os/guide/services/dhcp/disabling_dhcp_server/group.yml`


These should be corrected since they are quite simply, errors. I believe we both agree that these should be corrected.

__________

### Vague Titles (10)

In others, these titles are wrong because they apply to specific products or configuration steps, but are duplicated across different steps or across different types of objects:

- REUSED TITLE (All PKI Certificates Are Valid DoD Certificates): `fuse6/guide/fuse6/karaf/jboss_karaf-valid_dod_certificates.rule` and `fuse6/guide/fuse6/amq/jboss_activemq-pki_dod_certificates.rule`
- REUSED TITLE (Enable ExecShield): `rhel6/guide/system/permissions/restrictions/enable_execshield_settings/group.yml` and `rhel6/guide/system/permissions/restrictions/enable_execshield_settings/sysctl_kernel_exec_shield.rule`
- REUSED TITLE (Set Boot Loader Password): `rhel6/guide/system/accounts/accounts-physical/bootloader/group.yml` and `rhel6/guide/system/accounts/accounts-physical/bootloader/bootloader_password.rule`
- REUSED TITLE (Disable DHCP Client): `rhel6/guide/services/dhcp/disabling_dhcp_client/sysconfig_networking_bootproto_ifcfg.rule` and `rhel6/guide/services/dhcp/disabling_dhcp_client/group.yml`
- REUSED TITLE (Disable DNS Server): `rhel6/guide/services/dns/disabling_dns_server/group.yml` and `rhel6/guide/services/dns/disabling_dns_server/service_named_disabled.rule`
- REUSED TITLE (Record Attempts to Alter Logon and Logout Events): `linux_os/guide/system/auditing/auditd_configure_rules/audit_login_events/group.yml` and `linux_os/guide/system/auditing/auditd_configure_rules/audit_login_events/audit_rules_login_events.rule`
- REUSED TITLE (Enable ExecShield): `linux_os/guide/system/permissions/restrictions/enable_execshield_settings/group.yml` and `linux_os/guide/system/permissions/restrictions/enable_execshield_settings/sysctl_kernel_exec_shield.rule`
- REUSED TITLE (Set Boot Loader Password): `linux_os/guide/system/accounts/accounts-physical/bootloader/group.yml` and `linux_os/guide/system/accounts/accounts-physical/bootloader/bootloader_password.rule`
- REUSED TITLE (Disable DHCP Client): `linux_os/guide/services/dhcp/disabling_dhcp_client/sysconfig_networking_bootproto_ifcfg.rule` and `linux_os/guide/services/dhcp/disabling_dhcp_client/group.yml`
- REUSED TITLE (Disable DNS Server): `linux_os/guide/services/dns/disabling_dns_server/group.yml` and `linux_os/guide/services/dns/disabling_dns_server/service_named_disabled.rule`


These should also be corrected, as they duplicate the name from the group. This is more of a UX problem: as a user, two things which are not exactly equivalent shouldn't have the same name. Otherwise, it is ambiguous when referred to and it becomes much more likely to accidentally make a mistake (e.g., disabling a group when a rule was meant to be disabled). Also, with respect to IPv4 / IPv6 duplicate rules, having the IP layer information is important for users which to e.g., globally disable IPv6. w.r.t. cases where `default` vs `all` is used (in sysctl), they are referring to different configuration options which have different properties and thus duplication should be avoided.


Largely, I think it's for the best if, within a product, there are no duplicated titles. Especially considering, across the entire project (`ssg`) base, there are only 24 instances of duplicated titles. I'll be opening a PR when this one merges to fix the remaining cases and making it part of the build system.

Reply:
@redhatrises See comment at the bottom for response to "do not have to be unique".

re: this particular change, I'm actually in favor of removing this rule. The functionality of this rule is duplicated by the three other files in the directory (`audit_rules_login_events_*.rule`) and thus is redundant and not worth the effort of updating. Thoughts?